pkg/apis/apps/validation/validation.go

```diff
@@ -303,7 +303,7 @@ func ValidateDaemonSetStatusUpdate(ds, oldDS *apps.DaemonSet) field.ErrorList {
 }
 
 // ValidateDaemonSetSpec tests if required fields in the DaemonSetSpec are set.
-func ValidateDaemonSetSpec(spec *apps.DaemonSetSpec, fldPath *field.Path, opts apivalidation.PodValidationOptions) field.ErrorList {
+func ValidateDaemonSetSpec(spec *apps.DaemonSetSpec, fldPath *field.Path, opts apivalidation.PodValidationOptions, isLegacy bool /* TODO: after 1.22, remove isLegacy */) field.ErrorList {
```

Comment:
make this a field in PodValidationOptions that defaults safe (e.g. `AllowLongerDNSSearchPaths`), not a separate boolean field we have to plumb everywhere

populate the field in GetValidationOptionsFromPodSpecAndMeta based on the presence of more DNS search paths or longer search paths in the existing pod spec

Reply:
I will fix them all. Thanks!

Reply:
Done.

@aojea 
Cleaned up the code.
How about now?